grails-app/services/de/iteratec/osm/system/MissingJobResultCheckService.groovy

```diff
@@ -0,0 +1,103 @@
+package de.iteratec.osm.system
+
+import de.iteratec.osm.measurement.environment.wptserver.JobResultPersisterService
+import de.iteratec.osm.measurement.schedule.Job
+import de.iteratec.osm.result.JobResult
+import de.iteratec.osm.result.JobResultStatus
+import grails.gorm.transactions.Transactional
+import org.joda.time.DateTime
+import org.quartz.CronExpression
+
+@Transactional
+class MissingJobResultCheckService {
+    static final int FIRST_CHECK_DAYS_AGO = 1
+    static final long SCHEDULE_THRESHOLD_MINUTES = 3
+
+    JobResultPersisterService jobResultPersisterService
+
+    def fillMissingJobResults() {
+        Date latestCheckDate = getLatestCheck()?.date
+        int missingResults = 0
+        if(latestCheckDate) {
+            missingResults = fillMissingJobResultsSinceDate( new DateTime(latestCheckDate).minusMinutes(SCHEDULE_THRESHOLD_MINUTES.toInteger()))
+            new MissingJobResultCheck(date: new Date(), missingResults: missingResults).save()
+        }
+        else {
+            DateTime dateDaysAgo =  new DateTime().minusDays(FIRST_CHECK_DAYS_AGO)
+            missingResults = fillMissingJobResultsSinceDate( dateDaysAgo )
+            new MissingJobResultCheck(date: new Date(), missingResults: missingResults).save()
+        }
+        log.info("Fixed Missing JobResults: $missingResults")
+    }
+
+    private int fillMissingJobResultsSinceDate(DateTime since) {
+        int missingJobResults = 0
+        getPreparedJobList(since).forEach{ job ->
+            def existingResults = getJobResultsSince(job.job, job.since)
+            missingJobResults += checkForMissingJobResults(existingResults, job.since, job.job)
+        }
+        return missingJobResults
+    }
+
+    private List<Map> getPreparedJobList(DateTime since) {
+        return Job.findAllByDeletedAndActiveAndExecutionScheduleIsNotNull(false, true).collect {
+            ['job': it, 'id': it.id, 'since': since]
```

Comment:
Why do you include `since` in the map? As far as I can see, you only use it in a context where `since` is available anyway (otherwise you wouldn't be able to pass it as an argument to this function).
This way you can remove the whole `collect` and simply return a `List<Job>`.